ui/app/models/allocation.js

```diff
@@ -201,6 +205,27 @@ export default class Allocation extends Model {
     );
   }
 
+  @computed(
+    'hasStoppedRescheduling',
+    'node.status',
+    'resources.length',
+    'task.{config.errors.length,code.errors.length}'
+  )
+  get failureReason() {
+    switch (this.hasStoppedRescheduling) {
+      case this.node.status === 'failed':
+        return 'the node that the allocation was scheduled on failed.';
+      case this.resources.length === 0:
+        return 'the resources that the allocation was scheduled on were not available.';
+      case this.task.config.errors.length > 0:
+        return "the task or service's configuration was incorrect.";
+      case this.task.code.errors.length > 0:
+        return "there was an error in the task or service's code.";
```

Comment:
Will this prevent reschedule, or simply throw an error but otherwise continue on? Asking because I want to make sure that we're giving the right cause when we say "stopped .... because of ________"

Reply:
This is a derived state computation to show a reason why an `allocation` has stopped rescheduling to be shown in the `allocations.allocation.index` view. This is only shown when the current allocation (the one that's associated with the view) has stopped rescheduling.

Reply:
See the invocation in the template layer for usage details and you'll see that this only is applied for the top level allocation on the reschedule timeline because the rest of the items in the timeline are `RescheduleEventRow` which only provides links and no text.